tutorials/appengine-pusher/index.md

```diff
@@ -0,0 +1,449 @@
+                'id="chat_widget_member_' + member.id + '">'
+                + member.info.displayName + '</li>');
+            updateOnlineCount();
+        });
+        channel.bind('pusher:member_removed', function (member) {
+            // track member removals from channel
+            $('#chat_widget_member_' + member.id).remove();
+            updateOnlineCount();
+        });
+
+```
+
+### Bind to channel events
+
+[Event binding](https://pusher.com/docs/client_api_guide/client_events#bind-events) may be done bound to a channel
+or to all channels subscribed by a client.
+
+The following example shows attaching an event callback to a user triggered event.
+In the case of the chat application, the event callback will be used to update the messages 
+displayed by the chat application.
+
+[embedmd]:# (java/src/main/webapp/WEB-INF/view/chat.jsp /\/\/ bind to successful subscription/ /}\);/)
+```jsp
+// bind to successful subscription
+        channel.bind('pusher:subscription_succeeded', function (members) {
+            // receive list of members on this channel
+            var whosonline_html = '';
+            members.each(function (member) {
+                whosonline_html += '<li class="chat_widget_member" id="chat_widget_member_'
+                    +
+                    member.id + '">' + member.info.displayName + '</li>';
+            });
+```
+
+
+### Trigger server-side endpoint to send messages
+
+Now use a server-side endpoint as described earlier to trigger an event on a chat message
+ that is submitted.
+The client can be excluded from receiving the broadcast message by providing the socket ID.
+
+[embedmd]:# (java/src/main/webapp/WEB-INF/view/chat.jsp /\/\/ track socket_id/ /false;\n.*}\);/)
+```jsp
+// track socket_id to exclude recipient in subscription
+        socket_id = pusher.connection.socket_id;
+
+        // submit the message to /chat
+        $('#chat_widget_form').submit(function () {
+            var chat_widget_input = $('#chat_widget_input'),
+                chat_widget_button = $('#chat_widget_button'),
+                message = chat_widget_input.val(); //get the value from the text input
+            var data = JSON.stringify({
+                message: message,
+                channel_id: channel_name,
+                socket_id: socket_id
+            });
+            // trigger a server-side endpoint to send the message via Pusher
+            $.post('/message', data,
+                function (msg) {
+                    chat_widget_button.show(); //show the chat button
+                    if (msg.status == "SUCCESS") {
+                        chat_widget_input.val('');
+                        handleMessage(msg); //display the message
+                    } else {
+                        alert("Error sending chat message : " + msg.status);
+                    }
+                }, "json");
+
+            return false;
+        });
+```
+
+## Disconnecting from Pusher
```

Comment:
It isn't appearing for me now, but previously it appeared that all of the text after that code sample was in italics. It appears to be OK now though, so you should skip this issue.